pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -265,8 +299,138 @@ func (c *cyclonedxParser) GetPredicates(ctx context.Context) *assembler.IngestPr
+			justification = vexJustification
+		} else {
+			justification = model.VexJustificationNotProvided
+		}
+
+		publishedTime, _ = time.Parse(time.RFC3339, vulnerability.Published)
+		vd := model.VexStatementInputSpec{
+			Status:           status,
+			VexJustification: justification,
+			KnownSince:       publishedTime,
+			Statement:        vulnerability.Analysis.Detail,
+			StatusNotes:      fmt.Sprintf("%s:%s", string(status), string(justification)),
+		}
+
+		for _, affect := range *vulnerability.Affects {
+			vi := c.getAffectedPackages(ctx, vuln, vd, affect)
+			if vi == nil {
+				logger.Debugf("failed to get affected packages for vulnerability %s", vulnerability.ID)
+				continue
+			}
+			c.vulnData.vex = append(c.vulnData.vex, *vi...)
+
+			for _, v := range *vi {
+				if status == model.VexStatusAffected || status == model.VexStatusUnderInvestigation {
+					cv := assembler.CertifyVulnIngest{
+						Vulnerability: vuln,
+						VulnData: &model.ScanMetadataInput{
+							TimeScanned: publishedTime,
+						},
+						Pkg: v.Pkg,
+					}
+					c.vulnData.certifyVuln = append(c.vulnData.certifyVuln, cv)
+				}
+			}
+		}
+
+		for _, vulnRating := range *vulnerability.Ratings {
+			vm := assembler.VulnMetadataIngest{
+				Vulnerability: vuln,
+				VulnMetadata: &model.VulnerabilityMetadataInputSpec{
+					ScoreType:  model.VulnerabilityScoreType(vulnRating.Method),
+					ScoreValue: *vulnRating.Score,
+					Timestamp:  publishedTime,
+				},
+			}
+			c.vulnData.vulnMetadata = append(c.vulnData.vulnMetadata, vm)
+		}
+	}
+
+	return nil
+}
+
+// Get package name and range versions to create package input spec for the affected packages.
+func (c *cyclonedxParser) getAffectedPackages(ctx context.Context, vulnInput *model.VulnerabilityInputSpec, vexData model.VexStatementInputSpec, affectsObj cdx.Affects) *[]assembler.VexIngest {
+	logger := logging.FromContext(ctx)
+	pkgRef := affectsObj.Ref
+
+	// split ref using # as delimiter.
+	pkgRefInfo := strings.Split(pkgRef, "#")
+	if len(pkgRefInfo) != 2 {
+		logger.Debugf("[cdx vex] malformed affected-package reference: %q", affectsObj.Ref)
+		return nil
+	}
+	pkdIdentifier := pkgRefInfo[1]
+
+	// check whether the ref contains a purl
+	if strings.Contains(pkdIdentifier, "pkg:") {
+		pkg, err := asmhelpers.PurlToPkg(pkdIdentifier)
+		if err != nil {
+			logger.Debugf("[cdx vex] unable to create package input spec: %v", err)
+			return nil
+		}
+		c.identifierStrings.PurlStrings = append(c.identifierStrings.PurlStrings, pkdIdentifier)
+		return &[]assembler.VexIngest{{VexData: &vexData, Vulnerability: vulnInput, Pkg: pkg}}
+	}
+
+	if affectsObj.Range == nil {
+		logger.Debugf("[cdx vex] no range versions found for package ref %q", pkgRef)
+		return nil
+	}
+
+	var viList []assembler.VexIngest
+	for _, affect := range *affectsObj.Range {
+		// TODO: Handle package range versions (see - https://github.com/CycloneDX/bom-examples/blob/master/VEX/CISA-Use-Cases/Case-8/vex.json#L42)
```

Comment:
put a log here that currently it is not handled so that if someone uses it they know that it is missing?